src/Rings/mpolyquo-localizations.jl

```diff
@@ -2155,4 +2155,92 @@ function inverse(phi::MPolyQuoLocalizedRingHom)
   error("computation of inverse not implemented")
 end
 
+@doc raw"""
+    minimal_generating_set(I::MPolyLocalizedIdeal)
+
+Given an ideal `I` in the localization of a quotient of a  multivariate
+polynomial ring over a field at a point, return an array containing a
+minimal set of generators of `I`. If `I` is the zero ideal an empty list
+is returned.
+
+Note: This is only available for localizations at rational points. 
+"""
+@attr Vector{<:MPolyQuoLocRingElem}function minimal_generating_set(
+    I::MPolyQuoLocalizedIdeal{<:MPolyQuoLocRing{<:Field, <:Any,
+                                          <:Any, <:Any,
+                                          <:MPolyComplementOfKPointIdeal},
+                              <:Any,<:Any}
+  )
+  Q = base_ring(I)
+  !is_zero(I) || return typeof(zero(Q))[]
+  L = localized_ring(Q)
+
+  ## list of generators in the localized ring, append modulus
+  Jlist = lift.(gens(I))
+  nJlist = length(Jlist)
+  append!(Jlist,gens(modulus(Q)))
+
+  ## move to origing
+  shift, back_shift = base_ring_shifts(L)
+  I_shift = shifted_ideal(ideal(L,Jlist))
+  R = base_ring(I_shift)
+  oL = negdegrevlex(R)                    # the default local ordering
+
+
+  ## determine the relations
+  singular_assure(I_shift, oL)
+  syz_mod=Singular.syz(I_shift.gens.S)
+
+  ## prepare Nakayama-check for minimal generating system
+  F = free_module(R, length(Jlist))
+  Imax = ideal(R,gens(R))
+  M = sub(F,[F(syz_mod[i]) for i=1:Singular.ngens(syz_mod)])[1] + (Imax*F)[1]
+  oF =  negdegrevlex(R)*revlex(F)
+  res_vec = typeof(gen(I,1))[]
+
+  ## select by Nakayama
+  for i in 1:nJlist
+    if !(gen(F,i) in leading_module(M,oF))
+      append!(res_vec,[Q.(gen(I,i))])
+    end
+  end
+
+  return res_vec
+end
+
+@doc raw"""
+    small_generating_set(I::MPolyLocalizedIdeal)
```

Comment:
This docstring is misplaced. Also, the function below should probably just be a synonym for `minimal_generating_set`.

Reply:
No, the doctring has a Copy&Paste Error. The next commit amends this.
And no, this should not be an alias, because the other variant (which does it on foot)  is more clumsy and this one often suffices, if you do not require minimality, but only something reasonably small.